### PR TITLE
[admission-policy-engine] Added new checks for OperationPolicy 

### PR DESCRIPTION
**Pull request by libmonsoon-dev**
## Why do we need it, and what problem does it solve?
Close https://github.com/deckhouse/deckhouse/issues/9087

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: admission-policy-engine
type: feature
summary: Added ability to limit ingressClassName for Ingress and storageClassName for PersistentVolumeClaim for project / namespace / etc
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/static'...

File: modules/015-admission-policy-engine/templates/policies/operation-policy/constraint.yaml [hunk 1]
```diff
@@ -249,6 +255,47 @@ spec:
...
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: D8IngressClass
+metadata:
+  name: {{$cr.metadata.name}}
+  {{- include "helm_lib_module_labels" (list $context (dict "security.deckhouse.io/operation-policy" "")) | nindent 2 }}
+spec:
+  enforcementAction: {{ $cr.spec.enforcementAction | default "deny" | lower }}
+  match:
+    kinds:
+      - apiGroups: ["", "networking", "networking.k8s.io"]
```
Comment (miklezzzz) [inline modules/015-admission-policy-engine/templates/policies/operation-policy/constraint.yaml:271]: why did you decide to add "" apiGroup?

**Review by miklezzzz — APPROVED** [review]

File: modules/015-admission-policy-engine/templates/policies/operation-policy/constraint.yaml (hunk 1 above)
Comment (yalosev) [reply modules/015-admission-policy-engine/templates/policies/operation-policy/constraint.yaml:271]: And what is `networking` group?

File: modules/015-admission-policy-engine/templates/policies/operation-policy/constraint.yaml [hunk 2]
```diff
@@ -249,6 +255,47 @@ spec:
...
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: D8StorageClass
+metadata:
+  name: {{$cr.metadata.name}}
+  {{- include "helm_lib_module_labels" (list $context (dict "security.deckhouse.io/operation-policy" "")) | nindent 2 }}
+spec:
+  enforcementAction: {{ $cr.spec.enforcementAction | default "deny" | lower }}
+  match:
+    kinds:
+      - apiGroups: ["", "storage", "storage.k8s.io"]
```
Comment (yalosev) [inline modules/015-admission-policy-engine/templates/policies/operation-policy/constraint.yaml:292]: Same here, PVC is core/v1 resource, so only "" is enough

File: modules/015-admission-policy-engine/charts/constraint-templates/tests/ingress-class/constraint.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: D8IngressClass
+metadata:
+  name: test
+spec:
+  enforcementAction: "deny"
+  match:
+    kinds:
+      - apiGroups: ["", "networking", "networking.k8s.io"]
```
Comment (yalosev) [inline modules/015-admission-policy-engine/charts/constraint-templates/tests/ingress-class/constraint.yaml:9]:
```suggestion
      - apiGroups: ["networking.k8s.io"]
```

File: modules/015-admission-policy-engine/charts/constraint-templates/tests/storage-class/constraint.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: D8StorageClass
+metadata:
+  name: test
+spec:
+  enforcementAction: "deny"
+  match:
+    kinds:
+      - apiGroups: ["", "storage", "storage.k8s.io"]
```
Comment (yalosev) [inline modules/015-admission-policy-engine/charts/constraint-templates/tests/storage-class/constraint.yaml:9]:
```suggestion
      - apiGroups: [""]
```

File: modules/015-admission-policy-engine/templates/policies/operation-policy/constraint.yaml (hunk 1 above)
Comment (yalosev) [inline modules/015-admission-policy-engine/templates/policies/operation-policy/constraint.yaml:271]:
```suggestion
      - apiGroups: ["networking.k8s.io"]
```

File: modules/015-admission-policy-engine/templates/policies/operation-policy/constraint.yaml (hunk 2 above)
Comment (yalosev) [inline modules/015-admission-policy-engine/templates/policies/operation-policy/constraint.yaml:292]:
```suggestion
      - apiGroups: [""]
```

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Pull request merged**